poolqc/pooling_qc_sample.py

```diff
@@ -197,45 +203,32 @@ def Summary_table(run_alias, lane_alias):
     return columns, data, style_data_conditional, csv, download
 
 
-def update_sampleindices(run, rows, derived_virtual_selected_rows, colors):
+@app.callback(
+    dep.Output('SampleIndices', 'figure'),
+    [dep.Input('select_a_run', 'value'),
+     dep.Input('lane_select', 'value')])
+def update_sampleindices(run_alias, lane_alias):
+    run = df[(df['Run'] == run_alias) & (df['LaneNumber'] == lane_alias)]
+    run = run[~run['Run'].isna()].drop_duplicates('library')
+    run['Result'] = run['Coding Bases'] / run['Passed Filter Aligned Bases'] * 100
+    run['index'] = run['Index1'].str.cat(
+        run['Index2'].fillna(''), sep=' ')
+    run = run.sort_values('library')
+
+    total_RNA = len(run['library'])
+    Index_Pass = '%s/%s' % (sum(i > 20000 for i in run['SampleNumberReads']), total_RNA)
+    Index_Threshold = 50000000
     data = []
 
-    for inx, d in run.groupby(['index']):
+    for inx, d in run.groupby(['library']):
+        d['Threshold'] = Index_Threshold
+        d['Color'] = np.where((d['SampleNumberReads'] >= d['Threshold']), '#ffffff', '#db4b4b')
```

Comment:
Can the assignment to `d['Threshold']` be avoided by using `np.where((d['SampleNumberReads'] >= Index_Threshold), ...)`?

Reply:
The assignment was done for the threshold graph (`go.scatter`) as the values must be a list.